connaisseur/validators/notaryv1/notary.py

```diff
@@ -119,9 +114,38 @@ async def get_trust_data(
                         "www-authenticate"
                     ]
                 )
-                token = await self.__get_auth_token(session, auth_url)
-                return await self.get_trust_data(session, image, role, token)
+                repo_token = await self.__get_auth_token(session, auth_url)
+                root_trust_data, _ = await self.get_root_trust_data_and_auth(
+                    session, image, repo_token
+                )
+                return root_trust_data, repo_token
+            elif status == 401:
+                msg = (
+                    "Unable to find authorization endpoint in"
+                    "'www-authenticate' header for notary {notary_name}."
+                )
+                raise NotFoundException(message=msg, notary_name=self.name)
+            else:
+                response.raise_for_status()
+                data = await response.text()
+                return TrustData(json.loads(data), "root"), None
```

Comment:
nit: this conceptually "swallows" the repo_token

If I call get_x_and_token(a,b,token,c,d) I'd expect to be able to update my `token` with the returned token, whereas that's not possible here. That flow will never be hit, as we never call it from a higher level with a token, but potential for bugs during refactoring

Reply:
Hm, but changing the code to reflect your expectation would mean loosing the benefits we wanted to achieve  (no extra network call) or do I get this wrong? Do you think the function name could be more appropriate? If so, do you have suggestions? (`get_root_trust_data_and_auth_token_if_not_present` :grin: )

Reply:
```suggestion
                return TrustData(json.loads(data), "root"), repo_token
```

would be my suggestion. Basically, if you hand in a token, just return it, thus the return value is always a valid token. I don't think this would break any of the things you wrote, while "fixing" the function's "signature"